### PR TITLE
net/dns: support GetBaseConfig on Darwin OSS tailscaled

### PR DESCRIPTION
**Pull request by agottardo**
Updates tailscale/tailscale#177

It appears that the OSS distribution of `tailscaled` is currently unable to get the current system base DNS configuration, as GetBaseConfig() in manager_darwin.go is unimplemented. This PR adds a basic implementation that reads the current values in `/etc/resolv.conf`, to at least unblock DNS resolution via Quad100 if `--accept-dns` is enabled.

### CODE REVIEW

File: net/dns/manager_darwin.go [hunk 1]
```diff
@@ -83,8 +87,56 @@ func (c *darwinConfigurator) SetDNS(cfg OSConfig) error {
...
+	lines := strings.Split(string(resolvConf), "\n")
+	for _, line := range lines {
+		if len(line) == 0 {
+			continue
+		}
+		if line[0] == '#' {
+			continue
+		}
+		if strings.HasPrefix(line, "nameserver ") {
+			ip := strings.TrimSpace(line[11:])
```
Comment (bradfitz) [inline net/dns/manager_darwin.go:111]: CutPrefix, then you can drop the literal `11` too

File: net/dns/manager_darwin.go [hunk 2]
```diff
@@ -83,8 +87,56 @@ func (c *darwinConfigurator) SetDNS(cfg OSConfig) error {
...
+			ipAddr, err := netip.ParseAddr(ip)
+			if err != nil {
+				c.logf("ignoring invalid nameserver value '%q' in /etc/resolv.conf: %v", ip, err)
```
Comment (bradfitz) [inline net/dns/manager_darwin.go:114]: just %q, no need for quotes around it

File: net/dns/manager_darwin.go [hunk 3]
```diff
@@ -83,8 +87,56 @@ func (c *darwinConfigurator) SetDNS(cfg OSConfig) error {
...
+				continue
+			}
+			if ipAddr == tsaddr.TailscaleServiceIP() {
```
Comment (bradfitz) [inline net/dns/manager_darwin.go:117]: IPv6 one too

File: net/dns/manager_darwin.go [hunk 4]
```diff
@@ -83,8 +85,38 @@ func (c *darwinConfigurator) SetDNS(cfg OSConfig) error {
...
+		if ns == tsaddr.TailscaleServiceIP() || ns == tsaddr.TailscaleServiceIPv6() {
+			// If we find Quad100 in /etc/resolv.conf, we should ignore it
+			c.logf("ignoring 100.100.100.100 resolver IP found in /etc/resolv.conf")
+			continue
+		}
+		cfg.Nameservers = append(cfg.Nameservers, ns)
+	}
+	for _, sd := range resolvConf.SearchDomains {
+		cfg.SearchDomains = append(cfg.SearchDomains, sd)
+	}
```
Comment (andrew-d) [inline net/dns/manager_darwin.go:112]:
```suggestion
	cfg.SearchDomains = append(cfg.SearchDomains, resolvConf.SearchDomains...)
```

**Review by andrew-d — APPROVED** [review]

File: net/dns/manager_darwin.go (hunk 4 above)
Comment (andrew-d) [reply net/dns/manager_darwin.go]: (or just assign it, I guess? `cfg.SearchDomains = resolvConf.SearchDomains`?)

Comment (agottardo) [reply net/dns/manager_darwin.go]: Done.

**Pull request merged**